pkg/log/logger_test.go

```diff
@@ -0,0 +1,244 @@
+	RootLogger().Info("log.Logger() performance", zap.Int64("info (ns/op)", nsOp))
+}
+
+func benchmarkLegacyLoggerInfo(iterations int) int64 {
+	_ = Logger()
+	initTestLogger()
+	defer resetTestLogger()
+	start := time.Now()
+	for i := 0; i < iterations; i++ {
+		RootLogger().Info("test", zap.String("foo", "bar"))
+	}
+	return (time.Since(start).Nanoseconds()) / int64(iterations)
+}
+
+func BenchmarkScopedLoggerDebug(b *testing.B) {
+	benchmarkScopedLoggerDebug(b.N)
+}
+
+func TestScopedLoggerDebug(t *testing.T) {
+	nsOp := benchmarkScopedLoggerDebug(iterations)
+	Log(Test).Info("log.Log(...) performance (root=INFO)", zap.Int64("debug (ns/op)", nsOp))
+}
+
+func benchmarkScopedLoggerDebug(iterations int) int64 {
+	_ = Logger()
+	initTestLogger()
+	defer resetTestLogger()
+	UpdateLoggingConfig(map[string]string{
+		"log.level": "INFO",
+	})
+	start := time.Now()
+	for i := 0; i < iterations; i++ {
+		Log(K8Shim).Debug("test", zap.String("foo", "bar"))
+	}
+	return (time.Since(start).Nanoseconds()) / int64(iterations)
+}
+
+func BenchmarkScopedLoggerInfo(b *testing.B) {
+	benchmarkScopedLoggerInfo(b.N)
+}
+
+func TestScopedLoggerInfo(t *testing.T) {
+	nsOp := benchmarkScopedLoggerInfo(iterations)
+	Log(Test).Info("log.Log(...) performance (root=INFO)", zap.Int64("info (ns/op)", nsOp))
+}
+
+func benchmarkScopedLoggerInfo(iterations int) int64 {
+	_ = Logger()
+	initTestLogger()
+	defer resetTestLogger()
+	UpdateLoggingConfig(map[string]string{
+		"log.level": "INFO",
+	})
+	start := time.Now()
+	for i := 0; i < iterations; i++ {
+		Log(K8Shim).Info("test", zap.String("foo", "bar"))
+	}
+	return (time.Since(start).Nanoseconds()) / int64(iterations)
+}
+
+func BenchmarkScopedLoggerDebugEnabled(b *testing.B) {
+	benchmarkScopedLoggerDebugEnabled(b.N)
+}
+
+func TestScopedLoggerDebugEnabled(t *testing.T) {
+	nsOp := benchmarkScopedLoggerDebugEnabled(iterations)
+	Log(Test).Info("log.Log(...) performance (root=DEBUG)", zap.Int64("debug (ns/op)", nsOp))
+}
+
+func benchmarkScopedLoggerDebugEnabled(iterations int) int64 {
+	_ = Logger()
+	initTestLogger()
+	defer resetTestLogger()
+	UpdateLoggingConfig(map[string]string{
+		"log.test.level": "DEBUG",
+	})
+	start := time.Now()
+	for i := 0; i < iterations; i++ {
+		Log(Test).Debug("test", zap.String("foo", "bar"))
+	}
+	return (time.Since(start).Nanoseconds()) / int64(iterations)
+}
+
+func BenchmarkScopedLoggerInfoFiltered(b *testing.B) {
+	benchmarkScopedLoggerInfoFiltered(b.N)
+}
+
+func TestScopedLoggerInfoFiltered(t *testing.T) {
+	nsOp := benchmarkScopedLoggerInfoFiltered(iterations)
+	Log(Test).Info("log.Log(...) performance (root=DEBUG)", zap.Int64("info (ns/op)", nsOp))
+}
+
+func benchmarkScopedLoggerInfoFiltered(iterations int) int64 {
+	_ = Logger()
+	initTestLogger()
+	defer resetTestLogger()
+	UpdateLoggingConfig(map[string]string{
+		"log.test.level": "DEBUG",
+	})
+	start := time.Now()
+	for i := 0; i < iterations; i++ {
+		Log(K8Shim).Info("test", zap.String("foo", "bar"))
```

Comment:
I suppose the message "test" is unrelated to logger named "test".

Reply:
The message is purposely identical so that there's no variation in potential processing time (i.e. make sure it's not what we log but how we log it). The logger is chosen to determine which logging path we use relative to the config given.